src/plugins/intel_gpu/include/intel_gpu/runtime/debug_configuration.hpp

```diff
@@ -13,10 +13,34 @@
     auto stage_prof = cldnn::instrumentation::profiled_stage<primitive_inst>(\
         !cldnn::debug_configuration::get_instance()->dump_profiling_data.empty(), *this, stage)
 #define GPU_DEBUG_PROFILED_STAGE_CACHE_HIT(val) stage_prof.set_cache_hit(val)
+
+// Verbose log levels:
+// 0 - Silent mode (Default)
+// 1 - Minimal verbose:
+//     * May log basic info about device, plugin configuration, model and execution statistics
+//     * Mustn't log any info that depend on neither number of iterations or number of layers in the model
+//     * Minimal impact on both load time and inference time
+// 2 - Graph optimization verbose:
+//     * Includes info from level=1
+//     * May log info about applied graph transformations, memory allocations and other model compilation time steps
+//     * May impact compile_model() execution time
+//     * Minimal impact on inference time
+// 3 - Basic execution time verbose
+//     * Includes info from level=2
+//     * May log info during model execution
+//     * May log short info about primitive execution
+//     * May impact network execution time
+// 4 - Max verbosity
+//     * Includes info from level=3
+//     * May log any stage and print detailed info about each execution step
```

Comment:
Nice refactoring. I've always thought we need some clean-up here..

Some comments:
* what about having a macro for log level? Something like INFO=1, LOG=2, TRACE=3, TRACE_DETAIL=4
* what about unifying log fomat? Something like below. It will make it easy to parse the log.

```
// Raw log api
#define GPU_DEBUG_LOG_RAW(min_verbose_level) \
    if (verbose >= min_verbose_level) \
       std::cout << prefix << ":" <<

// Auto-prefix log api (this will be the default version)
#define GPU_DEBUG_LOG(min_verbose_level) \
    GPU_DEBUG_LOG_RAW(min_verbose_level) << ":" << __FILE__ << ":" << __LINE__
```

Reply:
I also though about adding special macro for log levels, but need to come up with good names. I think LOG=2 doesn't highlight when it can be used which may lead to regular misuse. 
Maybe INFO, GRAPH, INFER, TRACE or something like that?

Reply:
`GRAPH` and `INFER` sounds different usage, not the log level. I think we may introduce additional flag for the usage.
Then, what about INFO=1, INFO_DETAIL=2, TRACE=3, TRACE_DETAIL=4? (This doesn't look clean, but clear to understand..)

Reply:
If we separate use cases and log levels, then we can keep something shorter like `INFO, LOG, TRACE, DEBUG` or your first version (`INFO=1, LOG=2, TRACE=3, TRACE_DETAIL=4`) and add separate use case macro: 
```cpp
#define GPU_DEBUG_LOG_RAW(min_log_level) ...

#define GPU_DEBUG_LOG_INFO GPU_DEBUG_LOG_RAW(LOG_LEVEL_INFO)
#define GPU_DEBUG_LOG_TRACE GPU_DEBUG_LOG_RAW(LOG_LEVEL_TRACE)
..

#define GPU_DEBUG_LOG_GRAPH GPU_DEBUG_LOG_INFO 
#define GPU_DEBUG_LOG_INFER GPU_DEBUG_LOG_TRACE
```
How do you think?

Reply:
OK, sounds good to me